Revise target file related terminology

lukpueh opened the pull request:
- Update link to python package glossary (used to point to outdated PEP 426).
- Remove terms "projects", "releases" and "distributions" from definitions sections and refer the reader to the PPA glossary.
- Add definition for "distribution file" and clarify that it means "distribution package" as defined in the PPA glossary and may be referred to by the single word "distribution" in  this document.
- Add definition for target files.
- Use *distribution* or *distribution file*, or *target file*, instead of *package* throughout the document.

Commit:
Revise target file related terminology

- Update link to python package glossary (used to point to outdated
  PEP 426).
- Remove terms "projects", "releases" and "distributions" from
  definitions sections and refer the reader to the PPA glossary.
- Add definition for "distribution file" and clarify that it
  means "distribution package" as defined in the PPA glossary
  and may be referred to by the single word "distribution" in
  this document.
- Add definition for target files.
- Use distribution or distribution file, and in some cases target
  file, instead of package throughout the document.

lukpueh commented:
Reviews may focus on the changes in the Definitions section.

lukpueh commented:
Fixes #57

joshuagl commented on pep-0458.txt:
```diff
@@ -82,35 +82,35 @@ possibility of security breaches and prepare PyPI accordingly because it is a
 It was therefore easy to launch a man-in-the-middle attack between pip and
-PyPI, and change package content arbitrarily.  As a result, users could be tricked into
-installing malicious packages.  After the wiki
+PyPI, and change distribution content arbitrarily.  As a result, users could be tricked into
```
```suggestion
PyPI, and arbitrarily change the content of distributions.  As a result, users could be tricked into
```

joshuagl commented on pep-0458.txt:
```diff
@@ -82,35 +82,35 @@ possibility of security breaches and prepare PyPI accordingly because it is a
 against them [14]_.  Specifically, testing was done to see how pip would
 respond to these attacks with and without TUF.  Attacks tested included replay
-and freeze, arbitrary packages, slow retrieval, and endless data.  The post
+and freeze, arbitrary distribution, slow retrieval, and endless data.  The post
```
isn't "arbitrary packages" a name/class of attack from the relevant paper? If not several of these attack names might want to be made more descriptive to the reader

```suggestion
and freeze, arbitrary distribution files, slow retrieval, and endless data.  The post
```

lukpueh replied:
You're right, "A look in the mirror" calls this attack "arbitrary package", but the spec calls it "arbitrary installation", and the reference implementation "arbitrary software installation". Let's use the spec term. :)

joshuagl commented on pep-0458.txt:
```diff
@@ -154,41 +154,42 @@ This PEP focuses only on integrating TUF into PyPI. However, the reader is
+* Distribution file: A versioned archive file that contains Python packages,
+  modules, and other resource files that are used to distribute a release. In
+  this document a distribution file may also be referred to with the single
+  word distribution.
```
Worth pointing out that this is equivalent to the term "Distribution Package" in the Glossary of the Python Packaging User Guide?

lukpueh replied:
I briefly considered that. But it seemed more confusing than helpful.

An alternative would be to keep using the words "distribution" and "package" as before and adopt PyPI's definition of "distribution package", which also says that both words can be used alone to say the same thing.

joshuagl commented on pep-0458.txt:
```diff
@@ -154,41 +154,42 @@ This PEP focuses only on integrating TUF into PyPI. However, the reader is
-  provide.
+* Target files: As a rule of thumb, target files are all files on PyPI whose
+  integrity should be guaranteed with TUF. Typically, this includes
+  distribution files, and PyPI metadata such as simple indices.
```
```suggestion
  distribution files and PyPI metadata, such as simple indices.
```

joshuagl commented on pep-0458.txt:
```diff
@@ -260,20 +261,20 @@ Integrating PyPI with TUF
+might prevent users from installing or updating distributions, including pip itself,
 via TUF. Therefore, pip SHOULD provide an option e.g.,
-`--unsafely-disable-package-verification`, in order to work around such issues
+`--unsafely-disable-distribution-verification`, in order to work around such issues
```
I prefer we continue to refer to "package" in the option. The term "distribution" isn't used at all in Pip's `--help` output, but the term packages occurs 9 times.

lukpueh replied:
You have a point. But it seems weird if we almost completely ban the word "package" from this document.

joshuagl commented on pep-0458.txt:
```diff
@@ -1084,9 +1085,9 @@ information must be validated:
-3. If the packages themselves may have been tampered with, they can be
-   validated using the stored hash information for packages that existed at the
-   time of the last period.
+3. If the target files themselves may have been tampered with, they can be
```
It feels disorientating to refer to target files and distributions in the same sentence when discussing the same thing

```suggestion
3. If the distribution files themselves may have been tampered with, they can be
```

lukpueh replied:
Makes sense. But I suggest to use "target files" twice then, because this sentence should also apply to non-distribution file target files, e.g. index files, right?

lukpueh committed:
Refine distribution file definition

Acknowledge that it may be used interchangeably with "distribution
package", as it is defined in the PPA glossary, or simply "distribution"
or "package", for brevity, or because it is more common, e.g. for a
pip option.

Co-Authored-By: Joshua Lock <jlock@vmware.com>

trishankatdatadog commented:
LGTM to me on cursory read. @joshuagl are you happy?

joshuagl reviewed (approved):
Yes, looks good to me.

lukpueh commented:
Thanks!
